studio/components/interfaces/Organization/BillingSettingsV2/Subscription/DowngradeModal.tsx

```diff
@@ -31,36 +40,67 @@ const DowngradeModal = ({ visible, selectedTier, onClose, onConfirm }: Downgrade
             </p>
           </Alert>
 
-          {/* [Joshen] Improvement area: Show which projects might be most affected by the downgrade? */}
-          {/* [Joshen] Improvement area: Show which projects addons will be removed */}
-
           <ul className="mt-4 space-y-5 text-sm">
-            <li className="flex gap-3">
-              <IconMinusCircle />
-              <span>Add ons from all projects under this organization will be removed.</span>
-            </li>
-
             <li className="flex gap-3">
               <div>
                 <IconPauseCircle />
               </div>
-              <span>
-                Your projects will be paused if it exceeds the free plan quotas, or after a week of
-                inactivity
-              </span>
+              <span>Projects will be paused after a week of inactivity</span>
+            </li>
+
+            <li>
+              <div className="flex gap-3 mb-2">
+                <div>
+                  <IconMinusCircle />
+                </div>
+                <span>Add ons from all projects under this organization will be removed.</span>
+              </div>
+              {(subscription?.project_addons.length ?? 0) > 0 && (
+                <InformationBox
+                  title={`Warning: A total of ${subscription?.project_addons.length} project(s) will be affected from the downgrade`}
+                  description={
+                    <ul className="space-y-1 max-h-[100px] overflow-y-auto">
+                      {subscription?.project_addons.map((project) => {
+                        const needsRestart = project.addons.find(
+                          (addon) => addon.type === 'compute_instance'
+                        )
+                        const addons = project.addons.map((addon) => {
+                          if (addon.type === 'compute_instance')
+                            return `${addon.variant.name} Compute Instance`
+                          return addon.variant.name
+                        })
+                        return (
+                          <li key={project.ref} className="list-disc ml-6">
+                            {project.name}: {addons.join(', ')} will be removed.
+                            {needsRestart ? (
+                              <>
+                                {' '}
+                                Project will also{' '}
+                                <span className="text-amber-900">need to be restarted</span> due to
+                                change in compute instance
+                              </>
+                            ) : (
+                              ''
+                            )}
+                          </li>
+                        )
```

Comment:
This can be a component so we don't have the nested `.map()` :) (even if it's in the same file)